### ISSUE TITLE
ERRO[2454] Failed to cleanup overlay mounts

### ISSUE THREAD

**Issue by jamescassell** [containers/podman#8234]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

1. Install Fedora 33

2. Create source directory for overlay mount

3. Run container using overlay mount

4. Terminate container and observe error message

**Describe the results you received:**

```
[user@localhost ~]$ rm -rf ~/.local/share/containers/
[user@localhost ~]$ rm -rf ~/.local/share/yum/
[user@localhost ~]$ install -d -m 755 ~/.local/share/yum
[user@localhost ~]$ time podman run -v ~/.local/share/yum:/var/cache/dnf:z --rm fedora:33 yum makecache
Trying to pull registry.fedoraproject.org/fedora:33...
Getting image source signatures
Copying blob 8fde7942e775 done
Copying config 79fd58dc76 done
Writing manifest to image destination
Storing signatures
Fedora 33 openh264 (From Cisco) - x86_64        3.2 kB/s | 2.5 kB     00:00
Fedora Modular 33 - x86_64                      1.0 MB/s | 3.3 MB     00:03
Fedora Modular 33 - x86_64 - Updates            896 kB/s | 918 kB     00:01
Fedora 33 - x86_64 - Updates                    4.4 MB/s |  10 MB     00:02
Fedora 33 - x86_64                              3.3 MB/s |  72 MB     00:21
Metadata cache created.

real    1m51.497s
user    0m11.338s
sys     0m4.103s
[user@localhost ~]$ time podman run -v ~/.local/share/yum:/var/cache/dnf:O --rm fedora:33 echo test
test
ERRO[0000] Failed to cleanup overlay mounts for 0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2: failed to cleanup overlay /home/user/.local/share/containers/storage/overlay-containers/0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2/userdata/overlay directory: unlinkat /home/user/.local/share/containers/storage/overlay-containers/0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2/userdata/overlay/485125055/merge/updates-0e22a1f5a0a34771/repodata: directory not empty
ERRO[0000] cleanup storage: error removing container 0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2 root filesystem: 2 errors occurred:
        * unlinkat /home/user/.local/share/containers/storage/overlay-containers/0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2/userdata/overlay/485125055/merge/updates-0e22a1f5a0a34771/repodata: directory not empty
        * unlinkat /home/user/.local/share/containers/storage/overlay-containers/0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2/userdata/overlay/485125055/merge/updates-0e22a1f5a0a34771/repodata: directory not empty

ERRO[0000] Error removing container 0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2: error cleaning up container 0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2: error unmounting container 0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2 storage: failed to cleanup overlay /home/user/.local/share/containers/storage/overlay-containers/0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2/userdata/overlay directory: unlinkat /home/user/.local/share/containers/storage/overlay-containers/0894839b91233e719e017b8f432775e911c07e17f5cea0a68a78115b94ab8ab2/userdata/overlay/485125055/merge/updates-0e22a1f5a0a34771/repodata: directory not empty

real    0m1.030s
user    0m0.274s
sys     0m0.417s
[user@localhost ~]$
```

**Describe the results you expected:**

Clean Exit

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
(paste your output here)
[user@localhost ~]$ podman version
Version:      2.1.1
API Version:  2.0.0
Go Version:   go1.15.2
Built:        Wed Oct  7 12:21:20 2020
OS/Arch:      linux/amd64

```

**Output of `podman info --debug`:**

```
(paste your output here)

[user@localhost ~]$ podman info --debug
host:
  arch: amd64
  buildahVersion: 1.16.1
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.21-3.fc33.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.21, commit: 0f53fb68333bdead5fe4dc5175703e22cf9882ab'
  cpus: 2
  distribution:
    distribution: fedora
    version: "33"
  eventLogger: journald
  hostname: localhost.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.8.15-301.fc33.x86_64
  linkmode: dynamic
  memFree: 237228032
  memTotal: 1900122112
  ociRuntime:
    name: crun
    package: crun-0.15-5.fc33.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.15
      commit: 56ca95e61639510c7dbd39ff512f80f626404969
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.4-4.dev.giteecccdb.fc33.x86_64
    version: |-
      slirp4netns version 1.1.4+dev
      commit: eecccdb96f587b11d7764556ffacfeaffe4b6e11
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.0
  swapFree: 365162496
  swapTotal: 459272192
  uptime: 25h 23m 53.38s (Approximately 1.04 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.2.0-1.fc33.x86_64
      Version: |-
        fusermount3 version: 3.9.3
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.9.3
        using FUSE kernel interface version 7.31
  graphRoot: /home/user/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 1
  runRoot: /run/user/1000/containers
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 2.0.0
  Built: 1602087680
  BuiltTime: Wed Oct  7 12:21:20 2020
  GitCommit: ""
  GoVersion: go1.15.2
  OsArch: linux/amd64
  Version: 2.1.1


```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(paste your output here)
[user@localhost ~]$ rpm -q podman
podman-2.1.1-12.fc33.x86_64

```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Physical host, or Hyper-V VM same effect. Fedora 32 or 33, same effect.

**Comment by nalind** [comment]
podman 2.1.1 looks like it was using the overlay package from buildah 1.16.1, which didn't have an equivalent for https://github.com/containers/buildah/pull/2725 in it yet.  This will be fixed in a version of podman that uses buildah 1.17 or the not-yet-tagged 1.16.6.

**Comment by vrothberg** [comment]
It's fixed in the master branch and will be released with Podman v2.2.

Thanks for opening the issue, @jamescassell !
